leases/management/commands/import_winter_leases_from_csv.py

```diff
@@ -0,0 +1,171 @@
+import csv
+import logging
+from collections import namedtuple
+from datetime import date
+
+from dateutil.utils import today
+from django.core.exceptions import ValidationError
+from django.core.management import BaseCommand
+
+from customers.exceptions import MultipleProfilesException, NoProfilesException
+from customers.models import Boat, CustomerProfile
+from customers.services import ProfileService
+from leases.enums import LeaseStatus
+from leases.models import WinterStorageLease
+from resources.models import BoatType, WinterStoragePlace
+
+logger = logging.getLogger(__name__)
+
+LeaseInput = namedtuple(
+    "LeaseInput",
+    (
+        "section_id",
+        "place_number",
+        "name",
+        "email",
+        "phone",
+        "boat_width",
+        "boat_length",
+        "boat_register",
+        "comment",
+    ),
+)
+
+
+class Command(BaseCommand):
+    profile_service: ProfileService
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--profile-token",
+            nargs="?",
+            type=str,
+            help="[Required] The API token for Profile",
+        )
+        parser.add_argument(
+            "--lease-file-path",
+            nargs="?",
+            type=str,
+            help="[Required] The file to parse",
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            default=False,
+            help="Only show a list of the invoices that can be sent "
+            "and how many customer are missing phones in our system",
+        )
+
+    def get_or_create_customer_profile(
+        self, name: str, email: str = None, phone: str = None
+    ) -> CustomerProfile:
+        names = name.split(" ")
+        first_name = names.pop().capitalize()
+        last_name = " ".join([name.capitalize() for name in names])
+
+        try:
+            helsinki_profile = self.profile_service.find_profile(
+                first_name, last_name, email, phone, force_only_one=True
+            )
+            profile = CustomerProfile.objects.get(id=helsinki_profile.id)
+        except (NoProfilesException, CustomerProfile.DoesNotExist):
+            logger.debug(f"Creating profile for {last_name}, {first_name} ({email})")
+            profile = self.profile_service.create_profile(
+                first_name, last_name, email, phone
+            )
+        return profile
+
+    def handle(
```

Comment:
How should transactions be managed?
Changes to the profile service (profiles created) can not be rolled back.
Maybe it's still better to run in a transaction: profiles created during previous (failed) runs will be found in subsequent runs.

Reply:
> Changes to the profile service (profiles created) can not be rolled back
But with the `get_or_create`, we make sure that if the profile was created on a previous run, it will still be found.


> How should transactions be managed?
I'll add the transaction for each processed row 👍